dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKUtilities.java

```diff
@@ -865,4 +874,81 @@ static void assertServerHappy(LibertyServer server) throws Exception {
+        } catch (Exception e) {
+            Log.info(c, "getDetectedShortNames", "Bootstrap.getInstance() failed. This will likely cause the current repeat to skip " + e.toString());
+            detectedShortNames = null; //if bootstrap is broken we've got bigger problems, but we can avoid caching an exception
+            return new HashSet<String>();
+        }
+
+        String installRoot = b.getValue("libertyInstallPath");
+        String libPath = installRoot + "/lib/features/";
+
+        PathMatcher matcher = FileSystems.getDefault().getPathMatcher("glob:*.mf");
+
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get(libPath))) {
+            for (Path path : stream) {
+                if (!Files.isDirectory(path) && matcher.matches(path.getFileName())) {
+                    Files.lines(path)
+                                    .filter(line -> line.startsWith("IBM-ShortName: "))
+                                    .map(line -> line.replaceAll("IBM-ShortName: ", ""))
+                                    .forEach(detectedShortNames::add);
+                }
+            }
+        } catch (IOException e) {
+            Log.info(c, "getDetectedShortNames", "IO exception while reading .mf Files. This will likely cause the current repeat to skip " + e.toString());
+            detectedShortNames = null; //avoid caching an exception
+            return new HashSet<String>();
+        }
+
+        return detectedShortNames;
+
+    }
+
+    /**
+     * Checks if a feature or features are present in wlp/lib/features by reading the mf files
+     * and checking if all the provided IBM-ShortName are present.
+     *
+     * @param  featureSet A FeatureSet, all the names returned by getFeatures() will be checked for a matching IBM-ShortName in a mf file
+     * @return            true if all provided IBM-ShortName can be found in a .mf file in wlp/lib/features.
+     */
+    public static boolean areFeaturesMissing(FeatureSet featureSet) {
+        return areFeaturesMissing(featureSet.getFeatures());
+    }
+
+    /**
+     * Checks if a feature or features are present in wlp/lib/features by reading the mf files
+     * and checking if all the provided IBM-ShortName are present.
+     *
+     * @param  featuresToCheck IBM-ShortNames that are required to be available on the server
+     * @return                 true if all provided IBM-ShortName can be found in a .mf file in wlp/lib/features.
+     */
+    public static boolean areFeaturesMissing(Set<String> featuresToCheck) {
+
+        final String m = "areFeaturesPresent";
+        Log.info(c, m, "checking for " + String.join(", ", featuresToCheck));
+        Set<String> shortNames = getDetectedShortNames();
+
+        for (String s : featuresToCheck) {
+            boolean result = shortNames.contains(s);
+            if (!result) {
+                Log.info(c, m, "checked for " + s + " found " + result);
+                return true;
+            }
+        }
```

Comment:
Upon further reading, it's better than I thought.

You showed be the following log sample which is pretty clear.
```
repeatWithPredicate            I enter. Testing EE11_FEATURES_MicroProfile_70, EE10_FEATURES_MicroProfile_70
areFeaturesMissing             I checking for jsonb-3.0, mpHealth-4.0, restfulWSClient-4.0, mpReactiveStreams-3.0, mpTelemetry-2.0, mpFaultTolerance-4.1, mpConfig-3.1, mpJwt-2.1, mpRestClient-4.0, restfulWS-4.0, jsonp-2.1, mpReactiveMessaging-3.0, mpGraphQL-2.0, cdi-4.1, microProfile-7.0, mpOpenAPI-4.0, mpMetrics-5.1, mpContextPropagation-1.3
areFeaturesMissing             I checked for cdi-4.1 found false
areFeaturesMissing             I checking for jsonb-3.0, mpHealth-4.0, restfulWSClient-3.1, mpReactiveStreams-3.0, mpTelemetry-2.0, mpFaultTolerance-4.1, mpConfig-3.1, mpJwt-2.1, mpRestClient-4.0, restfulWS-3.1, jsonp-2.1, mpReactiveMessaging-3.0, mpGraphQL-2.0, cdi-4.0, microProfile-7.0, mpOpenAPI-4.0, mpMetrics-5.1, mpContextPropagation-1.3
areFeaturesMissing             I found all features
repeatWithPredicate            I found the following aceptable FeatureSets EE10_FEATURES_MicroProfile_70
```

I think ideally we would list all the features we didn't find, rather than just the first one, but otherwise this looks good.

e.g.
```
areFeaturesMissing             I missing features: cdi-4.1, restfulWS-4.0, restfulWSClient-4.0
```